### PR TITLE
feature/initial-guesses

### PR DESCRIPTION
**Pull request by aaustin141**
A variety of methods for generating initial guesses for iterative linear solvers.

### CODE REVIEW

**Comment by tcew** [comment]
@aaustin141  this looks good to me.  The extra tests added mean that this does not reduce code coverage.

@noelchalmers please approve so that we can squash and merge.

File: solvers/elliptic/src/ellipticSettings.cpp [hunk 1]
```diff
@@ -66,6 +66,29 @@ void ellipticAddSettings(settings_t& settings,
...
+
+  settings.newSetting(prefix+"INITIAL GUESS EXTRAP DEGREE",
+                      "-1",
+                      "Degree used for EXTRAP initial guess schemes.");
+
+  settings.newSetting(prefix+"INITIAL GUESS EXTRAP COEFFS METHOD",
+                      "MINNORM",
+                      "Method for selecting coefficients with EXTRAP initial guess schemes.",
+                      {"MINNORM", "CPQR"});
+
```
Comment (noelchalmers) [inline solvers/elliptic/src/ellipticSettings.cpp:91]: Is using the initial guess machinery really part of the elliptic solver? I.e., should these settings be here, or should they be part of the larger solver (ins, fpe) that would construct them?

**Review by noelchalmers — CHANGES_REQUESTED** [review]

File: solvers/elliptic/src/ellipticSettings.cpp (hunk 1 above)
Comment (aaustin141) [reply solvers/elliptic/src/ellipticSettings.cpp:91]: Yeah, that's probably better, but there's a disadvantage to that:  if we were to add a new initial guess method, we would have to update `*Settings.cpp` in all the individual solvers to make it available to them.

Probably the right solution would be for the linear solver settings to be "owned" by `linearSolver_t` instead.

Comment (noelchalmers) [reply solvers/elliptic/src/ellipticSettings.cpp:91]: In other instances where I find myself adding the same settings over and over again, I usually make an `*AddSettings` in the lib/namespace which I can reuse. Something like the parAlmond or elliptic AddSettings may work.

**Review by noelchalmers — APPROVED** [review]

**Pull request merged**